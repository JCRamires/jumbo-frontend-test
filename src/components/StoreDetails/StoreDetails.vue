<script setup lang="ts">
import cx from "classnames";
import { JumboStore } from "../../store/types";
import GoogleMapsIFrame from "../GoogleMapsFrame";

interface Props {
  store: JumboStore;
}

const { store } = defineProps<Props>();

const sundayLabelClasses = cx("font-semibold", store.sundayOpen ? "text-green-500" : "text-red-500")
</script>

<template>
  <h2>Store details</h2>
  <div class="grid grid-cols-[max-content_minmax(0,_1fr)] gap-2">
    <div class="font-bold justify-self-end">City:</div>
    <div>
      {{ store.city }}
    </div>
    <div class="font-bold justify-self-end">Address:</div>
    <div>
      <div>{{ store.street }}, {{ store.street2 }}</div>
      <div :v-if="store.street3">{{ store.street3 }}</div>
    </div>
    <div class="font-bold justify-self-end">Postal code:</div>
    <div>{{ store.postalCode }}</div>
  </div>

  <h2>Working hours</h2>
  <div>
    From <span class="font-semibold">{{ store.todayOpen }}</span> to
    <span class="font-semibold text-green">{{ store.todayClose }}</span>
  </div>
  <div>
    <span :class="sundayLabelClasses">{{
      store.sundayOpen ? "Opens" : "Closes"
    }}</span>
    on sundays
  </div>

  <h2>Location</h2>
  <GoogleMapsIFrame :latitude="store.latitude" :longitude="store.longitude" />
</template>
