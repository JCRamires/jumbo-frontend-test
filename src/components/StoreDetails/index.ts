import StoreDetails from "./StoreDetails.vue";

export default StoreDetails;
