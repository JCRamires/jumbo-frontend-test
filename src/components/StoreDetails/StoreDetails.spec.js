import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import StoreDetails from "./StoreDetails.vue";

const JUMBO_STORE = {
  type: "StoreListRO",
  uuid: "V7cKYx4X0QUAAAFMTmYM5CXj",
  addressName: "Jumbo Alkmaar Duijvelshoff",
  street: "Muiderwaard",
  street2: "416",
  street3: "",
  city: "Alkmaar",
  postalCode: "1824 XT",
  distance: 0,
  todayOpen: "07:00",
  todayClose: "22:00",
  latitude: "52.645601",
  longitude: "4.749492",
  locationType: "SupermarktPuP",
  newStore: 0,
  collectionPoint: true,
  complexNumber: "33282",
  sapStoreID: "3574",
  favourite: false,
  isHomeStore: false,
  showWarningMessage: true,
  sundayOpen: true,
};

describe("StoreDetails", () => {
  it("mounted component should match snapshot", async () => {
    const wrapper = mount(StoreDetails, {
      props: {
        store: JUMBO_STORE,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
