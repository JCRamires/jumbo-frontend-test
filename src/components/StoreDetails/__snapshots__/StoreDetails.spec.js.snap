// Vitest Snapshot v1

exports[`StoreDetails > mounted component should match snapshot 1`] = `
<div
  data-v-app=""
>
  
  <h2>
    Store details
  </h2>
  <div
    class="grid grid-cols-[max-content_minmax(0,_1fr)] gap-2"
  >
    <div
      class="font-bold justify-self-end"
    >
      City:
    </div>
    <div>
      Alkmaar
    </div>
    <div
      class="font-bold justify-self-end"
    >
      Address:
    </div>
    <div>
      <div>
        Muiderwaard, 416
      </div>
      <div
        v-if=""
      />
    </div>
    <div
      class="font-bold justify-self-end"
    >
      Postal code:
    </div>
    <div>
      1824 XT
    </div>
  </div>
  <h2>
    Working hours
  </h2>
  <div>
     From 
    <span
      class="font-semibold"
    >
      07:00
    </span>
     to 
    <span
      class="font-semibold text-green"
    >
      22:00
    </span>
  </div>
  <div>
    <span
      class="font-semibold text-green-500"
    >
      Opens
    </span>
     on sundays 
  </div>
  <h2>
    Location
  </h2>
  <div
    class="iframe-container"
    data-v-a826f365=""
  >
    <iframe
      data-v-a826f365=""
      frameborder="0"
      height="600"
      marginheight="0"
      marginwidth="0"
      scrolling="no"
      src="https://maps.google.com/maps?q=52.645601,4.749492&z=14&output=embed"
      width="800"
    >
      
    
    </iframe>
  </div>
  
</div>
`;
